client/components/tinymce/plugins/mentions/suggestion.jsx

```diff
@@ -3,74 +3,67 @@
-		fullName: React.PropTypes.string,
-		query: React.PropTypes.string
-	},
-
-	getDefaultProps() {
-		return {
-			avatarUrl: '',
-			username: '',
-			fullName: '',
-			query: ''
-		};
-	},
-
-	getRegExpFor( type, textToHighlight ) {
-		const expressions = {};
-
-		expressions.username = '(^' + textToHighlight + ')(\\w*)\\s*';
-		expressions.fullName = '(^.*?)(\\b' + textToHighlight + ')(.*)';
-
-		return new RegExp( expressions[ type ], 'ig' );
-	},
-
-	highlight( content, textToHighlight, type ) {
-		const matcher = this.getRegExpFor( type, textToHighlight ),
-			matches = matcher.exec( content );
-
-		if ( matches ) {
-			const highlights = [];
-			let highlighted = false;
-
-			for ( let i = 1, length = matches.length; i < length; i++ ) {
-				let item = matches[ i ];
-
-				if ( textToHighlight.toLowerCase() === item.toLowerCase() && ! highlighted ) {
-					item = <strong className="mentions__highlight" key={ i }>{ matches[ i ] }</strong>;
-					highlighted = true;
-				}
-
-				highlights.push( item );
-			}
+const getRegExpFor = function( type, textToHighlight ) {
+	const expressions = {};
 
-			return highlights;
-		}
+	expressions.username = '(^' + textToHighlight + ')(\\w*)\\s*';
+	expressions.fullName = '(^.*?)(\\b' + textToHighlight + ')(.*)';
+
+	return new RegExp( expressions[ type ], 'ig' );
+};
+
+const highlight = function( content, textToHighlight, type ) {
+	const matcher = getRegExpFor( type, textToHighlight ),
+		matches = matcher.exec( content );
 
-		return [ content ];
-	},
+	if ( matches ) {
+		const highlights = [];
+		let highlighted = false;
 
-	render() {
-		const { avatarUrl, username, fullName, query } = this.props,
-			highlightedUsername = this.highlight( username, query, 'username' ),
-			highlightedFullName = this.highlight( fullName, query, 'fullName' );
+		for ( let i = 1, length = matches.length; i < length; i++ ) {
+			let item = matches[ i ];
 
-		highlightedUsername.unshift( '@' );
+			if ( textToHighlight.toLowerCase() === item.toLowerCase() && ! highlighted ) {
+				item = <strong className="mentions__highlight" key={ i }>{ matches[ i ] }</strong>;
+				highlighted = true;
+			}
+
+			highlights.push( item );
+		}
 
-		return (
-			<div className="mentions__suggestion">
-				<img className="mentions__avatar" src={ avatarUrl } />
-				<span className="mentions__username">{ highlightedUsername }</span>
-				<small className="mentions__fullname">{ highlightedFullName }</small>
-			</div>
-		);
+		return highlights;
 	}
-} );
+
+	return [ content ];
+};
+
+const Suggestion = ( { avatarUrl, username, fullName, query } ) => {
+	const highlightedUsername = highlight( username, query, 'username' );
+	const highlightedFullName = highlight( fullName, query, 'fullName' );
+
+	highlightedUsername.unshift( '@' );
+
+	return (
+		<div className="mentions__suggestion">
+			<img className="mentions__avatar" src={ avatarUrl } />
+			<span className="mentions__username">{ highlightedUsername }</span>
```

Comment:
Instead of adding an `@` to the string here, would it be valuable to reuse the `<EditorMention />` component instead?